components/dashboard/cards/contactLink.tsx

```diff
@@ -0,0 +1,166 @@
+import { AppDispatch } from '@/lib/redux/store'
+import {
+    InputLabel,
+    FormControl,
+    TextField,
+    SelectChangeEvent
+} from '@mui/material'
+
+import { Close, Edit, Check } from '@mui/icons-material'
+import { enqueueSnackbar } from 'notistack'
+import bannerUtils from '@/utils/banners'
+
+import {
+    setContactLinks,
+    updateContactLinks
+} from '@/lib/redux/slices/contactLinks'
+import {
+    deleteContactLink,
+    updateContactLink
+} from '@/lib/db/cms/contact-links'
+import { ConfirmDeleteButton, CircleIconButton } from '@/components/buttons'
+import IconSelectField from '@/components/form/fields/IconSelectField'
+import { StyledContactLinkCard } from './StyledCards'
+
+const { statuses, messages } = bannerUtils
+
+const ContactLinkCard = ({ field }: any) => {
+    const dispatch = useDispatch<AppDispatch>()
+    const [isEditMode, setIsEditMode] = useState<boolean>(false)
+    const [iconValue, setIconValue] = useState<string>('')
+    const [linkValue, setLinkValue] = useState<string>('')
+    const { id, icon, link } = field
+
+    //Functions for Delete Button
+    const onDeleteContactLink = async () => {
+        try {
+            const contactLinkData = await deleteContactLink(id)
+            dispatch(setContactLinks(contactLinkData))
+            enqueueSnackbar('Contact Link Deleted', {
+                variant: statuses.SUCCESS
+            })
+        } catch (error) {
+            enqueueSnackbar(messages.TECHNICAL_DIFFICULTIES, {
+                variant: statuses.ERROR
+            })
+        }
+    }
+
+    //Functions for Edit Button
+    const onEditModeChange = () => {
+        setLinkValue(link)
+        setIconValue(icon)
+        setIsEditMode(!isEditMode)
+    }
+    const onEditModeConfirm = async () => {
+        try {
+            await updateContactLink({
+                link: linkValue,
+                icon: iconValue,
+                id: id
+            })
+            dispatch(
+                updateContactLinks({ link: linkValue, icon: iconValue, id: id })
+            )
+            setIsEditMode(false)
+            enqueueSnackbar('Contact Link Updated', {
+                variant: statuses.SUCCESS
+            })
+        } catch (error) {
+            enqueueSnackbar(messages.TECHNICAL_DIFFICULTIES, {
+                variant: statuses.ERROR
+            })
+        }
+    }
+    const onEditModeCancel = () => {
+        setLinkValue(link)
+        setIconValue(icon)
+        setIsEditMode(false)
+    }
+
+    //Functions for event changes on select and text fields. Hi Philly
+    const onIconSelectChange = (event: SelectChangeEvent) => {
+        setIconValue(event.target.value as string)
+    }
+    const onLinkChange = (event: { target: { value: string } }) => {
+        setLinkValue(event.target.value)
+    }
+    return (
+        <StyledContactLinkCard>
+            <div className="link-card">
+                <div className="card-header">
```

Comment:
So i see you have a lot of checks here for `isEditMode`. To clean up this return and make this spicy, you should do this:
1. Create another 2 components within this folder called `EditIconForm` and `AddIconForm` or something like that. 
2. The return can look like this (paraphrasing):
```
if (isEditMode) {
 return <EditIconForm />
} 

return <AddIconForm />
```
And the Edit Icon form can have the edit & delete functions all live within there. The add icon form can have the necessary functions for that as well. Compartmentalizing the code instead of having it all live in one file, with like 9 if/else checks everywhere. You make 1 check for what kind of mode you're on, then let the sub components handle themselves

Reply:
i attempted something